cmd/kube-apiserver/app/server.go

```diff
@@ -299,7 +299,18 @@ func Run(s *options.ServerRunOptions) error {
 	}
 
 	admissionControlPluginNames := strings.Split(s.GenericServerRunOptions.AdmissionControl, ",")
-	pluginInitializer := kubeadmission.NewPluginInitializer(client, sharedInformers, apiAuthorizer)
+	var cloudConfigReader *os.File
+
+	if s.CloudProvider.CloudConfigFile != "" {
+		cloudConfigReader, err = os.Open(s.CloudProvider.CloudConfigFile)
+		if err != nil {
+			glog.Errorf("Error opening cloud configuration file %s: %#v", s.CloudProvider.CloudConfigFile, err)
+		}
+		if cloudConfigReader != nil {
```

Comment:
Passing a reader is really ugly, I'd recommend you pass the bytes down.  Just because cloud config reader accepts it today doesn't mean it would continue.  This will be a lot cleaner as ReadAll(), and passing a buffer to the deep initializer.